torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -591,6 +592,15 @@ ProcessGroupNCCL::ProcessGroupNCCL(
             << "\nUSE_HIGH_PRIORITY_STREAM: "
             << options_->is_high_priority_stream
             << "\nNCCL_DEBUG: " << ncclDebugLevel;
+
+  if (ucc_lib == nullptr) {
+    ucc_lib = loadTorchUCC();
```

Comment:
There is still a race here where two threads could end up loading TorchUCC twice. You probably need to use something like `std::call_once`.

Reply:
fixed